pathtools/fs.go

```diff
@@ -160,3 +186,19 @@ func (m *mockFs) glob(pattern string) ([]string, error) {
 func (m *mockFs) Lstat(path string) (stats os.FileInfo, err error) {
 	return nil, errors.New("Lstat is not yet implemented in MockFs")
 }
+func (m *mockFs) ListDirsRecursive(name string) (dirs []string, err error) {
+	name = filepath.Clean(name)
+	dirs = append(dirs, name)
+	if name == "." {
+		name = ""
```

Comment:
I guess it's worth noting that this assumes that no absolute paths were entered into m.all. It seems to me that it'd be nice to either add an assertion confirming that whenever a path is added to m.all that it isn't absolute, or to update this to handle that case.

It looks like there's already an assertion in a test below - I'd just move it into the mockFs itself

Reply:
I fixed the tests to test the absolute paths too.

Reply:
Fixed again, MockFs globs will now only match an absolute path if given an absolute pattern.  The mock glob tests were actually finding the issue that you were describing, but I had incorrectly worked around it by using one MockFs that contained only relative paths and another that contained only absolute paths.